core/modules/ccontrol/UserQueryInvalid.h

```diff
@@ -0,0 +1,91 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_CCONTROL_USERQUERYINVALID_H
+#define LSST_QSERV_CCONTROL_USERQUERYINVALID_H
+
+// System headers
+#include <memory>
+#include <mutex>
+
+// Third-party headers
+
+// Qserv headers
+#include "ccontrol/UserQuery.h"
+#include "qdisp/MessageStore.h"
+#include "qmeta/types.h"
+
+// Forward decl
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+
+/// UserQueryInvalid : implementation of the UserQuery which is used
+/// to indicate invalid queries.
+class UserQueryInvalid : public UserQuery {
+public:
+
+    UserQueryInvalid(std::string const& message)
+        : _message(message),
+          _messageStore(std::make_shared<qdisp::MessageStore>()) {}
+
+    UserQueryInvalid(UserQueryInvalid const&) = delete;
+    UserQueryInvalid& operator=(UserQueryInvalid const&) = delete;
+
+    // Accessors
```

Comment:
I was lazy and did not want to create separate .cc file for this class so I made all its methods inline (they are all empty so it does not take much space in header). In other classes these methods have some non-trivial implementations and are defined in .cc files.